MapboxNavigation/Style.swift

```diff
@@ -209,6 +210,115 @@ public class ResumeButton: UIControl {
+    }
+    let paragraphStyle = NSMutableParagraphStyle()
+    let fontSize = UIFont.systemFontSize
+    let strings = NSMutableAttributedString()
+    let numberFormater = NumberFormatter()
+    let stackView = UIStackView()
+    
+    var speedLimit: SpeedLimit? {
+        didSet {
+            guard let speedLimit = speedLimit else { return }
+            guard let speedString = numberFormater.string(from: speedLimit.value as NSNumber) else { return }
+            
+            labels.removeAll()
+            
+            let defaultLabels = defaultTextSplit.map { (string: String) -> SpeedLimitLabel in
+                let label: SpeedLimitLabel = .forAutoLayout()
+                label.attributedText = NSMutableAttributedString(string: string.uppercased(), attributes: [
+                    .paragraphStyle: paragraphStyle,
+                    .font: UIFont.systemFont(ofSize: fontSize * 0.9)
+                    ])
+                return label
+            }
+            labels.append(contentsOf: defaultLabels)
+            
+            let speedLabel: SpeedLimitLabel = .forAutoLayout()
+            speedLabel.attributedText = NSMutableAttributedString(string: speedString, attributes: [
+                .font: UIFont.boldSystemFont(ofSize: fontSize * 2),
+                .paragraphStyle: paragraphStyle
+                ])
+            labels.append(speedLabel)
+            
+            let unitLabel: SpeedLimitLabel = .forAutoLayout()
+            unitLabel.attributedText = NSMutableAttributedString(string: "\n\(speedLimit.unit.localizedSpeedUnit)", attributes: [
+                .font: UIFont.boldSystemFont(ofSize: fontSize / 1.3),
+                .paragraphStyle: paragraphStyle
+                ])
+            labels.append(unitLabel)
+            
+            updateStackView(with: labels)
+        }
+    }
+    
+    @objc dynamic public var borderColor: UIColor? {
+        get {
+            guard let color = layer.borderColor else { return nil }
+            return UIColor(cgColor: color)
+        }
+        set {
+            layer.borderColor = newValue?.cgColor
+        }
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    
+    func commonInit() {
+        paragraphStyle.lineHeightMultiple = 0.9
+        paragraphStyle.alignment = .center
+        layoutMargins = SpeedLimitSign.textInsets
+        stackView.axis = .vertical
+        stackView.contentMode = .scaleAspectFill
+        stackView.distribution = .equalSpacing
+        addSubview(stackView)
+        stackView.layoutIfNeeded()
+    }
+    
+    func updateStackView(with labels: [SpeedLimitLabel]) {
+        stackView.subviews.forEach { $0.removeFromSuperview() }
+        stackView.addArrangedSubviews(labels)
+        stackView.layoutIfNeeded()
+    }
```

Comment:
Converted this to a stack view, and things are no longer really working. @JThramer @frederoni does anything look funny to you here?

Reply:
@bsudekum what do you mean?
>  things are no longer really working...